packages/playground/transform-plugin/vite.config.js

```diff
@@ -0,0 +1,22 @@
+let transformCount = 1
+
+const transformPlugin = {
+  name: 'transform',
+  transform(code, id) {
+    if (id === require.resolve('./index.js')) {
```

Comment:
Thanks for the PR @mattcompiles!

The Windows test issue is solved by using `normalizePath` when comparing paths to ids. In Vite, paths are normalized so plugins need to take this into account when comparing external paths.
```suggestion
const { normalizePath } = require('vite')
let transformCount = 1

const transformPlugin = {
  name: 'transform',
  transform(code, id) {
    if (id === normalizePath(require.resolve('./index.js'))) {
```

I still need to review how this affects the module graph, but at least you'll get all green ticks.

Reply:
Thanks for that. Makes sense.